jb/itch5/array_based_order_book.hpp

```diff
@@ -0,0 +1,617 @@
+#ifndef jb_itch5_array_based_order_book_hpp
+#define jb_itch5_array_based_order_book_hpp
+
+#include <jb/itch5/price_field.hpp>
+#include <jb/itch5/price_levels.hpp>
+#include <jb/config_object.hpp>
+#include <jb/feed_error.hpp>
+#include <jb/log.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <map>
+#include <sstream>
+#include <utility>
+#include <vector>
+
+namespace jb {
+namespace itch5 {
+
+/// A simple representation for price + quantity
+using half_quote = std::pair<price4_t, int>;
+
+/// The values used to represent the range of valid prices
+/// LOWEST_PRICE < valid price < HIGHEST_PRICE
+const price4_t LOWEST_PRICE = price4_t(0);
+const price4_t HIGHEST_PRICE = max_price_field_value<price4_t>();
+const int WIRE_MAX = HIGHEST_PRICE.as_integer();
```

Comment:
Should these be defined in the header for price?  There is nothing book specific about them.
In general we have not defined whether constants are supposed to be ALL_CAPS or not, I am fine with it, but we should add a defiition.

So far I have been using

    int const FOO = ...;

instead of 

   const int FOO = ...;

C++ types are more readable if the const-volatile-qualifiers are at the right of the type (because you always read types right-to-left in this case).  Please change the const qualifiers.